jaxrs-api/src/test/java/javax/ws/rs/core/EntityTagTest.java

```diff
@@ -0,0 +1,70 @@
+package javax.ws.rs.core;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Mockito.*;
+
+import javax.ws.rs.ext.RuntimeDelegate;
+
+public class EntityTagTest {
+
+    private EntityTag tag1;
+    private EntityTag tag2;
+    private EntityTag tag3;
+    private EntityTag tag4;
+
+
+    @Before
+    public void setUp() {
+        RuntimeDelegate.setInstance(mock(RuntimeDelegate.class));
+
+        tag1 = new EntityTag("value1", true);
+        tag2 = new EntityTag("value1", true);
+        tag3 = new EntityTag("value2", true);
+        tag4 = new EntityTag("value1", false);
```

Comment:
I don't want to be too niggling here, but what do you think about giving these fields more descriptive names? To understand the tests, you currently have to scroll up to the `setUp()` method to see how these `EntityTag` instances have been created, because the field names don't help much.

I personally would even go one step further and define these instances in each test method. Of course this means more code, but tests are much easier to understand.

Reply:
I think this is a question of personal taste. My person choice would be BDD style, with the typical //given section holding pretty clearly named variables instead of a setUp method at all.

Reply:
@chkal I do not see a downside. It is more a philosopical question: How to select that particular "false matches" that we want to cover with tests? Like "does not match a string", "does not match a subclass", and so on? We have to draw a line somewhere. OTOH I understand this particular test, but it more looks like "This class correctly implements 'equals'". Maybe we could find a generic solution for that test case, as it is independent of this particular class?

Reply:
> I think this is a question of personal taste. My personal choice would be BDD style, with the typical //given section holding pretty clearly named variables instead of a setUp method at all.

I agree. However, I'm not too strong on this one. We could also improve the overall test structure later.

@NiklasMehner: Sorry that all these basic discussions are delaying your pull request. But there is still much very basic stuff to discuss. And primarily working on GitHub is a pretty new concept for the spec.

Reply:
No, problem at all. 

Regarding the test selection:
Most of the time I'm trying to achieve 100% branch coverage. So if there is a branch like in this case:

if (!(obj instanceof EntityTag)) [...]

Both cases (expression evaluating to true and false) should be tested. I treat && and || as branches as well (they are in the bytecode).
If you then add some boder cases (null, invalid input) you imho get a reasonable set of tests.

A generic test for equals is not that easy to do. https://github.com/jqno/equalsverifier is a nice library for this, but I guess it is not worth adding a dependency just for testing a few equals-methods.

Also probably these classes can become (valhalla) value classes once we get to that kind of language level and then we won't need any equals methods any more.

I think the given/when/then is a bit overblown for tests this simple, but I will inline the fields and try to come up with some more descriptive names.